libs/src/sdk/oauth/OAuth.ts

```diff
@@ -140,25 +148,61 @@ export class OAuth {
     )
   }
 
-  login() {
-    if (!this.config.appName) {
+  async isWriteAccessGranted(params: IsWriteAccessGrantedRequest) {
+    const { userId, apiKey } = parseRequestParameters(
+      'isWriteAccessGranted',
+      IsWriteAccessGrantedSchema
+    )(params)
+    if (!this.apiKey && !apiKey) {
+      this._surfaceError(
+        'Need to init Audius SDK with API key or pass in API Key directly to oauth.isWriteAccessGranted.'
+      )
+    }
+    const authorizedApps = await this.config.usersApi.getAuthorizedApps({
+      id: userId
+    })
+
+    const foundIndex = authorizedApps.data?.findIndex(
+      (a) =>
+        a.address.toLowerCase() ===
+        `0x${(apiKey || this.apiKey)!.toLowerCase()}`
+    )
+    return foundIndex !== undefined && foundIndex > -1
+  }
+
+  login(scope: OAuthScope = 'read') {
+    if (!this.config.appName && !this.apiKey) {
       this._surfaceError('App name not set (set with `init` method).')
       return
     }
+    if (scope === 'write' && !this.apiKey) {
+      this._surfaceError(
+        'Need to init Audius SDK with API key in order to use oauth.login with scope = write'
```

Comment:
nit: maybe `'The \'write\' scope requires Audius SDK to be initialized with an API key'` or something?